hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/acl/iam/IamSessionPolicyResolver.java

```diff
@@ -0,0 +1,329 @@
+    final JsonNode effectNode = statement.get("Effect");
+    if (effectNode != null) {
+      if (effectNode.isTextual()) {
+        final String effect = effectNode.asText();
+        if (!"Allow".equalsIgnoreCase(effect)) {
+          throw new OMException("Unsupported Effect - " + effect, NOT_SUPPORTED_OPERATION);
+        }
+        return;
+      }
+
+      throw new OMException(
+          "Invalid Effect in JSON policy (must be a String) - " + effectNode, INVALID_REQUEST);
+    }
+
+    throw new OMException("Effect is missing from JSON policy", INVALID_REQUEST);
+  }
+
+  /**
+   * Reads a JsonNode and converts to a Set of String, if the node represents
+   * a textual value or an array of textual values.  Otherwise, returns
+   * an empty List.
+   */
+  private static Set<String> readStringOrArray(JsonNode node) {
+    if (node == null || node.isMissingNode() || node.isNull()) {
+      return Collections.emptySet();
+    }
+    if (node.isTextual()) {
+      return Collections.singleton(node.asText());
+    }
+    if (node.isArray()) {
+      final Set<String> set = new HashSet<>();
+      node.forEach(n -> {
+        if (n.isTextual()) {
+          set.add(n.asText());
+        }
+      });
+      return set;
+    }
+
+    return Collections.emptySet();
+  }
+
+  /**
+   * Parses and returns prefixes from Conditions (if any).  Also validates
+   * that if there is a Condition, there is only one and that the Condition
+   * operator and attribute are supported.
+   * <p>
+   * Only the StringEquals operator and s3:prefix attribute are supported.
+   */
+  private static Set<String> parsePrefixesFromConditions(JsonNode stmt) throws OMException {
+    Set<String> prefixes = Collections.emptySet();
+    final JsonNode cond = stmt.get("Condition");
+    if (cond != null && !cond.isMissingNode() && !cond.isNull()) {
+      if (cond.size() != 1) {
+        throw new OMException("Only one Condition is supported", NOT_SUPPORTED_OPERATION);
+      }
+
+      if (!cond.isObject()) {
+        throw new OMException(
+            "Invalid Condition (must have operator StringEquals " + "and attribute s3:prefix) - " +
+            cond, INVALID_REQUEST);
+      }
+
+      final String operator = cond.fieldNames().next();
+      if (!"StringEquals".equals(operator)) {
+        throw new OMException("Unsupported Condition operator - " + operator, NOT_SUPPORTED_OPERATION);
+      }
+
+      final JsonNode attribute = cond.get("StringEquals");
+      if ("null".equals(attribute.asText())) {
+        throw new OMException("Missing Condition attribute - StringEquals", INVALID_REQUEST);
+      }
+
+      if (!attribute.isObject()) {
+        throw new OMException("Invalid Condition attribute structure - " + attribute, INVALID_REQUEST);
+      }
+
+      final String attributeFieldName = attribute.fieldNames().hasNext() ? attribute.fieldNames().next() : null;
+      if (!"s3:prefix".equals(attributeFieldName)) {
```

Comment:
It's said that  "Context key names are not case-sensitive."

in https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition.html 

Can you double check it?